Objective-C/CBLDocument.mm

```diff
@@ -254,6 +252,9 @@ - (void) replaceC4Doc: (CBLC4Document*)c4doc {
 
 - (void) markAsRemoteDoc {
     CBL_LOCK(self) {
+        if (_c4Doc || _database)
+            [NSException raise: NSInternalInconsistencyException
+                        format: @"Document with c4doc cannot be marked as remote doc"];
```

Comment:
`4doc` is internal info. Maybe `Document cannot be saved as a remote doc` or `Document is not a remote doc`?